server/src/test/java/umm3601/user/UserControllerSpec.java

```diff
@@ -125,6 +135,39 @@ public void setupEach() throws IOException {
     userController = new UserController(db);
   }
 
+  /**
+   * Construct an instance of `ContextUtil`, which is essentially
+   * a mock context in Javalin. See `mockContext(String, Map) for
+   * more details.
+   */
+  private Context mockContext(String path) {
+    return mockContext(path, Map.of());
+  }
+
+  /**
+   * Construct an instance of `ContextUtil`, which is essentially a mock
+   * context in Javalin. We need to provide a couple of attributes, which is
+   * the fifth argument, which forces us to also provide the (default) value
+   * for the fourth argument. There are two attributes we need to provide:
+   *
+   *   - One is a `JsonMapper` that is used to translate between POJOs and JSON
+   *     objects. This is needed by almost every test.
+   *   - The other is `maxRequestSize`, which is needed for all the ADD requests,
+   *     since `ContextUtil` checks to make sure that the request isn't "too big".
+   *     Those tests fails if you don't provide a value for `maxRequestSize` for
+   *     it to use in those comparisons.
+   */
+  private Context mockContext(String path, Map<String, String> pathParams) {
+    return ContextUtil.init(
+        mockReq, mockRes,
+        path,
+        pathParams,
+        HandlerType.INVALID,
+        Map.ofEntries(
+          entry(io.javalin.plugin.json.JsonMapperKt.JSON_MAPPER_KEY, new JavalinJackson()),
+          entry(ContextUtil.maxRequestSizeKey, maxRequestSize)));
```

Comment:
What would you think about referencing this directly from JavalinConfig, rather than hardcoding it? It's a bit of extra work down here, but it's a bit DRYer and less brittle (and saves us from having that long discussion up at the top 🙂

(I agree with you that it would be nicer if `JavalinConfig` had a better API for these things 😛)

```suggestion
          // Use the default maximum request size.
          entry(ContextUtil.maxRequestSizeKey, new JavalinConfig().maxRequestSize)));
```

Reply:
In principle I would _totally_ prefer to refer to JavalinConfig. Unfortunately we'd have to construct an object, extract that value, and then make that available, all of which is hassle. I'd rather put that energy into just getting rid of this whole `ContextUtil` business so we don't need that constant at all.

Reply:
But of course I construct a `JavalinJackson` every time this gets called, which also makes no sense. I'm fixing that, and I'll move your suggestion so that it only happens once.

Reply:
So I pull this and my construction of `JavalinJackson` up to static fields so they should only happen once now.